### PR TITLE
Laboratory work #2, Tsvetkova Vladislava - 22FPL2

### PR DESCRIPTION
**Pull request by Vladays**

### CODE REVIEW

File: lab_2_tokenize_by_bpe/main.py [hunk 1]
```diff
@@ -16,16 +14,22 @@ def prepare_word(
...
+    if not isinstance(raw_word, str):
         return None
-    list_of_tokens = list(raw_word)
-    if end_of_word:
-        list_of_tokens.append(end_of_word)
-    if start_of_word:
-        list_of_tokens.insert(0, start_of_word)
-    return tuple(list_of_tokens)
+    if not (isinstance(start_of_word, str))\
+            and start_of_word is not None:
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:20]: strange check

File: lab_2_tokenize_by_bpe/main.py [hunk 2]
```diff
@@ -38,22 +42,21 @@ def collect_frequencies(
...
-    dict_frequencies = {}
-
-    splitted_text = text.split()
-    for i in set(splitted_text):
-        word = prepare_word(i, start_of_word, end_of_word)
-        if not word:
+    word_list = text.split()
+    freq_dict = {}
+    for word in word_list:
+        freq = word_list.count(word)
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:52]: count count function for the same values many times - try to accelerate

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_tokenize_by_bpe/main.py (hunk 1 above)
Comment (Vladays) [reply lab_2_tokenize_by_bpe/main.py]: improved

File: lab_2_tokenize_by_bpe/main.py (hunk 2 above)
Comment (Vladays) [reply lab_2_tokenize_by_bpe/main.py]: improved

**Pull request closed without merging**